### PR TITLE
feat: pubsub responder

### PR DESCRIPTION
**Pull request by stephhuynh18**
In the mode pubsub responder, it will listen to pubsub query messages and create an ipfs task to respond with the anchor commit

### CODE REVIEW

File: src/services/ipfs-service.ts [hunk 1]
```diff
@@ -174,4 +193,84 @@ export class IpfsService implements IIpfsService {
...
+  async stop(): Promise<void> {
+    if (this.pubsub$) {
+      this.pubsub$.unsubscribe()
+    }
+
+    await this.ipfs.pubsub.unsubscribe(this.pubsubTopic)
+  }
+
+  private async createPubsub() {
+    if (this.respondToPubsubQueries) {
```
Comment (stbrody) [inline src/services/ipfs-service.ts:206]: if you invert the condition here you can early return and avoid an indentation layer which will help with readability.

File: src/services/ipfs-service.ts [hunk 2]
```diff
@@ -174,4 +193,84 @@ export class IpfsService implements IIpfsService {
...
+      // We have to subscribe to pubsub to keep ipfs connections alive.
+      // TODO Remove this when the underlying ipfs issue is fixed
+      await this.ipfs.pubsub.subscribe(this.pubsubTopic, () => {
+        /* do nothing */
+      })
+    }
+  }
+
+  async handleMessage(message: PubsubMessage): Promise<void> {
+    if (message.typ === MsgType.QUERY) {
```
Comment (stbrody) [inline src/services/ipfs-service.ts:250]: again, early return can avoid an indentation layer and increase readability.

**Review by stbrody — APPROVED** [review]
lgtm, but I'd like Sergey to take a look as well, especially at the rxjs plumbing

**Review by ukstv — APPROVED** [review]
Huge

**Pull request merged**